pkg/controller/compute/address.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package compute
+
+import (
+	"context"
+	"time"
+
+	"github.com/crossplane/provider-gcp/pkg/clients/address"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/api/compute/v1"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane/provider-gcp/apis/compute/v1beta1"
+	gcp "github.com/crossplane/provider-gcp/pkg/clients"
+)
+
+// Error strings.
+const (
+	errNotAddress = "managed resource is not an Address"
+)
+
+// SetupAddress adds a controller that reconciles Address managed resources.
+func SetupAddress(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, poll time.Duration) error {
```

Comment:
I haven't reviewed the whole PR yet (sorry about that, trying to find some time in these last days of the year), but I want to recommend you to update your branch with the newest changes on main. That's because we changed the Setup method signature (#380). You can see [this example](https://github.com/crossplane/provider-gcp/blob/df5cb191f5c45856991b3f7e5a15d030c5be3217/pkg/controller/compute/globaladdress.go#L52) to check how it is right now.

Reply:
Will do. I wondered why CI build suddenly started to fail, but then realized that there was another change somewhere in progress.

Reply:
Hmm, seems _ SetupContainerRegistry_ was missed when updating to the latest _Setup_ signature.

Reply:
Ok, updated the _Setup_ signature for _Adress_ and _ContainerRegistry_.